app/codes/blockchain.py

```diff
@@ -135,29 +115,18 @@ def mine_block(self, cur, text):
         last_block = last_block_cursor.fetchone()
     #    con.close()
         previous_hash = last_block[0] if last_block is not None else 0
-        # print(last_block)
-        # previous_block = self.chain[-1]
-#		print(previous_block);
-        # previous_hash = self.hash(previous_block)
-        block = {'index': len(self.chain) + 1,
-                 'timestamp': str(datetime.datetime.now()),
-                 'proof': 0,
-                 'text': text,
-                 'previous_hash': previous_hash}
+        block = {
+            'index': get_last_block_index() + 1,
+            'timestamp': str(datetime.datetime.now()),
+            'proof': 0,
+            'text': text,
+            'previous_hash': previous_hash
+        }
 
-        proof = self.proof_of_work(block)
-        print("Current block proof: ", proof)
-#		print("previous_hash = ",previous_hash)
-        block = self.create_block(
-            cur, block['timestamp'], proof, previous_hash, text)
+        block_hash = self.proof_of_work(block)
```

Comment:
where is block['proof'] picked up for inclusion in the newly created block? This code seems to suggest that it is ignored and only hash taken.. won't new blocks continue to show the proof as None?

Reply:
That's done inside `proof_of_work` line 54
https://github.com/asqisys/newrl/pull/57/files#diff-ed54aa23cedfa5f57d8139cad9aa955e8c3987f769b958faab4eeda3857864a6R54

The block is passed by reference and any change to it inside the function will be reflected to the block object outside

Reply:
ok.